compiler/optimizer/ValuePropagationCommon.cpp

```diff
@@ -1289,6 +1428,136 @@ void OMR::ValuePropagation::transformArrayCopyCall(TR::Node *node)
       {
       TR::ResolvedMethodSymbol* methodSymbol = comp()->getMethodSymbol();
 
+#ifdef J9_PROJECT_SPECIFIC
+      if (doRuntimeNullRestrictedTest)
+         {
+         if (trace())
+            {
+            comp()->dumpMethodTrees("Trees before modifying for null restricted array check");
+            comp()->getDebug()->print(comp()->getOutFile(), comp()->getFlowGraph());
+            }
+         /*
+               ==== Before ===
+  _curTree---> n9n       treetop
+               n8n         call  java/lang/System.arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
+               n3n           aload  <parm 0 [LSomeInterface;>[#419  Parm]
+               n4n           iconst 0
+               n5n           aload  <parm 1 [LSomeInterface;>[#420  Parm]
+               n6n           iconst 0
+               n7n           iload  TestSystemArraycopy4.ARRAY_SIZE I[#421  Static]
+
+               ==== After ===
+               n48n      astore  <temp slot 3>[#429  Auto]
+               n3n         aload  <parm 0 [LSomeInterface;>[#419  Parm]
+               n50n      istore  <temp slot 4>[#430  Auto]
+               n4n         iconst 0
```

Comment:
I think with your recent change to only use temporaries for non-constant values, the stores of the constants should be removed from these comments.  Similarly, the commoning arrows shown below for `n4n` and `n6n` need to be removed, and the loads of `#430` and `#432` in the slow block should become `iconst 0` nodes.